lib/pbench/server/templates.py

```diff
@@ -272,45 +350,48 @@ def add_mapping(self, name: str, body: Dict[AnyStr, Any]):
 
     def resolve(self):
         """
-        Expand templates to create all the information necessary to register a
-        document template with Elasticsearch.
-
-        Some templates share settings JSON files, and the tool templates are
-        generated from a skeleton/base and tool "fragment"; all of this logic
-        is managed here.
+        In the initial implementation, this just loads the template files.
 
-        Once this method returns, the body() method can be called to return
-        the full Elasticsearch JSON payload necessary to register a document
-        template and its index pattern.
+        Placeholder: this will check the template DB and decide whether we
+        need to load the JSON files from disk based on modification dates.
         """
         self.load()
 
-        # FIXME: It's ugly reaching into the JsonFile object from here; this
-        # ought to be refactored into JsonFile somehow.
-        if self.tool:
-            tool_mapping = copy.deepcopy(self.base.mappings.json)
+    def generate_index_name(self, source, toolname=None):
+        """
+        Return a fully formed index name given its template, prefix, source
+        data (for an @timestamp field) and an optional tool name.
 
-            # This is ugly: we get the version from the tool-specific overlay;
-            # but that becomes a subdocument of the skeleton, and we need the
-            # version there. So we delete from one and create in the other.
-            del self.mappings.json["_meta"]
-            tool_mapping["_meta"] = dict(version=self.version)
-            sub_map = self.mappings.json
-            self.mappings.json = tool_mapping
-            self.add_mapping(self.name, sub_map)
+        Args:
 
```

Comment:
Having the "given" list not match the function parameter list is jarring.

Having an empty `Args:` section is worse.  ;-)

Reply:
Too much refactoring with deferred cleanup; `toolname` isn't even used at this level.